### PR TITLE
Add Offer Jetpack Complete page main component & layout.

### PR DESCRIPTION
**Pull request by elliottprogrammer**
#### Proposed Changes

This PR creates the a JetpackRnaDialogCard component that can be reused for similar RNA cards with an RNA image on the left and product content on the right.  The PR just creates the basic component and layout, whereas the content body will still need to be added in.
See Figma designs for final designs: R1U6Ww74mdgMrKUJ21BnTL-fi-3061%3A11111

Completes task: 1203759331596262-as-1203791007934498/f

**Desktop view:**

Screenshot on 2023-01-27 at 19-56-59

.
**Mobile View:**

Screenshot on 2023-01-27 at 19-58-32

#### Testing Instructions

* boot up this PR 
    * Run `git fetch && git checkout add/jp-complete-page-layout-component`
    * Run `yarn start`
    * Goto http://calypso.localhost:3000
* OR use the Live Link provided below
* Create a new JN site
* Navigate to `/jetpack/connect/plans/complete/[JN Site]`.  If using the Live Link, please append `?flags=jetpack/offer-complete-after-activation` to the url.
* Verify you can see the main Jetpack Complete component as shown in the screenshots below.
* If you have an unactivated product license key available, you should see the "Already have an existing plan or license key? Click here to get started" (desktop view). Or "Activate license" (mobile view).  Test that this message shows if a licence key is available and that it does not show if no license keys are available.
* View in mobile and desktop views. Compare with figma designs:  R1U6Ww74mdgMrKUJ21BnTL-fi-3061%3A11111

*

#### Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

Related to #

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - 0-as-1203791007934498

### CODE REVIEW

File: client/my-sites/plans/jetpack-plans/jetpack-complete-page/maybe-show-license-activation-link.tsx
Comment (TheSteveK) [inline client/my-sites/plans/jetpack-plans/jetpack-complete-page/maybe-show-license-activation-link.tsx]: This is nitpicky for sure.  But, I feel like MaybeShowLicenseActivationLink is a little vague when describing what this function does.  Maybe something more like ShowLicenseActivationLinkIfAvailable or Eligible or something like that?

Comment (elliottprogrammer) [reply client/my-sites/plans/jetpack-plans/jetpack-complete-page/maybe-show-license-activation-link.tsx]: @TheSteveK , thanks for the review! 👍 The activate license link text/copy is per the figma design and also it only displays to the user _If_, in fact, they have 1 or more available/unactivated licenses. With that said, I think it's clear & makes sense, but we surely could ask Design for another opinion if you still think it may cause user confusion. 🙂

Comment (elliottprogrammer) [reply client/my-sites/plans/jetpack-plans/jetpack-complete-page/maybe-show-license-activation-link.tsx]: Ohh!! You mean the naming of the function?..  Oh, yes sure, whatever you think. Sconce I am/have been AFK all week, I expected you or Andri would take ove this PR and make any changes you think and get it shipped. 😉👍

Comment (TheSteveK) [reply client/my-sites/plans/jetpack-plans/jetpack-complete-page/maybe-show-license-activation-link.tsx]: Yeah, sorry should have been more clear on what I was talking about.  Andrii is also out this week.  But I'll make the changes I suggested and see if I can get this driven today.

**Comment by elliottprogrammer** [comment]
Awesome, thanks @TheSteveK!  Yes, the function/component is more clear now. Thanks for the suggestion and changing it. 👍
Everything LGTM! We just need someones review approval and this PR can be shipped (I cannot approve my own PR). 😉

File: client/my-sites/plans/jetpack-plans/jetpack-complete-page/show-license-activation-link-if-available.tsx
Comment (andrii-lysenko) [inline client/my-sites/plans/jetpack-plans/jetpack-complete-page/show-license-activation-link-if-available.tsx]: this constant should be lowercase, however, I would even suggest removing it and simply use `!hasFetchedLicensesCounts` as it's only used in 2 places.

**Review by andrii-lysenko — APPROVED** [review]
Tested and looks good to me. Great job!

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7814446

Thank you @elliottprogrammer for including a screenshot in the description! This is really helpful for our translators.

File: client/components/jetpack/card/jetpack-rna-dialog-card/index.tsx
Comment (mattgawarecki) [inline client/components/jetpack/card/jetpack-rna-dialog-card/index.tsx:31]: I realize this was merged three days ago, but I noticed something while reviewing #73387 and wanted to bring it to y'all's attention:

`{ children && children }` doesn't accomplish anything different than `{ children }`, correct? This seems like a redundant boolean condition we could safely remove.

Comment (elliottprogrammer) [reply client/components/jetpack/card/jetpack-rna-dialog-card/index.tsx:31]: Yes totally, you're right, thank you for spotting this. I'll clean it up on a following PR. 👍

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7814446).

**Pull request merged**
